src/metabase/db/data_source.clj

```diff
@@ -10,6 +12,61 @@
   (:import java.sql.DriverManager
            java.util.Properties))
 
+(defn- h2? [url]
+  (.startsWith url "jdbc:h2:"))
+
+(defn- h2-path [url]
+  (let [[_ path] (re-matches #"jdbc:h2:file:(.*)$" url)]
+    path))
+
+(defn- h2-migration-paths [url]
+  (let [base (h2-path url)]
+    {:v1db   (str base ".mv.db")
+     :v2db   (str base ".v2.mv.db")
+     :script (str base ".v2migration.sql")}))
+
+(defn- h2-v1-db [url]
+  (let [src (.newInstance (Class/forName "org.h2_v1_4_197.jdbcx.JdbcDataSource"))]
+    (.setURL src url)
+    (.getConnection src)))
+
+(defn- h2-v2-db [url]
+  (let [src (.newInstance (Class/forName "org.h2.jdbcx.JdbcDataSource"))]
+    (.setURL src (str url ".v2"))
```

Comment:
This is a bit of a hack - is there a better place for this transparently added suffix?